src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeDeletionTest.java

```diff
@@ -0,0 +1,136 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.controllers;
+
+import com.google.common.base.Strings;
+import com.google.common.io.Resources;
+import com.microsoft.azure.storage.CloudStorageAccount;
+import com.microsoft.azure.storage.StorageCredentialsAccountAndKey;
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.StreamSupport;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import static com.jayway.awaitility.Awaitility.await;
+import static org.hamcrest.Matchers.is;
+
+public class EnvelopeDeletionTest {
+
+    private transient long scanDelay;
+
+    private transient String accountName;
+
+    private transient String testStorageKey;
+
+    private transient CloudBlobContainer testContainer;
+
+    @Before
+    public void setUp() throws Exception {
+        Config conf = ConfigFactory.load();
+        this.scanDelay = Long.parseLong(conf.getString("test-scan-delay"));
+        this.accountName = conf.getString("test-storage-account-name");
+        this.testStorageKey = conf.getString("test-storage-account-key");
+
+        StorageCredentialsAccountAndKey credentials =
+            new StorageCredentialsAccountAndKey(accountName, testStorageKey);
+
+        testContainer = new CloudStorageAccount(credentials, true)
+            .createCloudBlobClient()
+            .getContainerReference("test");
+    }
+
+
+    @Test
+    public void should_delete_zip_file_after_successful_ingestion() throws Exception {
+        List<String> files = Arrays.asList("1111006.pdf");
+        String metadataFile = "1111006.metadata.json";
+        String destZipFilename = getRandomFilename(null, "8_24-06-2018-00-00-00.zip");
+
+        uploadZipFile(files, metadataFile, destZipFilename); // valid zip file
+
+        await()
+            .atMost(scanDelay + 10000, TimeUnit.MILLISECONDS)
+            .until(() -> storageHasFile(destZipFilename), is(false));
+        Assert.assertThat("File has not been deleted.",
+            storageHasFile(destZipFilename), is(false));
```

Comment:
we normally use AssertJ accross the projects

Reply:
done